File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.py
def divisor(num):
    temp = 0
    for i in range(2, int(num**(1/2))+1):
        if num % i == 0:
            #temp.append(i)
            temp += 2
            break
    # print(temp)
    return temp


def solution(n):
    answer = 0
    for i in range(2, n+1):
        temp = divisor(i)
        if temp != 2:
            #answer.append(i) 
            answer += 1 

    return answer

File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        double e = 0;
        boolean c = true;
        
        for(int i = 2;i<=n;i++){
            e=Math.sqrt(i);
            if(i==2||i==3){
                answer++;
                continue;
            }
            for(int j = 2;j<=e;j++){
                if(i%j==0){
                    c=false;
                    break;
                }
            }
            if(c==true)
                answer++;
            c=true;
        }
        return answer;
    }
}